src/main/clojure/clara/rules/compiler.clj

```diff
@@ -424,18 +422,29 @@
                                              [name (list accessor '?__fact__)])
                                            accessors)))
 
-        token-assignments (mapcat #(list (symbol (name %)) (list 'get-in '?__token__ [:bindings %])) binding-keys)
+        ;; Binding keys that are not newly created are drawn from the token.
+        token-binding-keys (remove element-bindings (variables-as-keywords constraints))
```

Comment:
Yeah, we could explicitly compute this and add it as another field to the beta graph. If doing so makes things easier to debug (i.e., the small additional complexity added to the beta graph is justified), then it's worth pursuing, but I agree that is out of scope here.